activation/nipost.go

```diff
@@ -149,17 +150,13 @@ func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.PoetC
 		nb.persist()
 	}
 
-	// Phase 1: receive proofs from PoET services
+	// Phase 1: query PoET services for proofs
 	if nb.state.PoetProofRef == nil {
-		select {
-		case <-time.After(time.Until(poetProofDeadline)):
-		case <-ctx.Done():
-			return nil, 0, ctx.Err()
-		}
-		poetProofRef := nb.getBestProof(ctx, challengeHash)
+		getProofsCtx, cancel := context.WithDeadline(ctx, poetProofDeadline)
+		defer cancel()
+		poetProofRef := nb.getBestProof(getProofsCtx, challengeHash)
```

Comment:
the name challengeHash is getting confusing for me.
sometimes it's Hash32, sometimes it's []byte (in the struct)
why not simply use []byte everywhere?

Reply:
You are right, it should be unified. I'd go with a `Hash32` as it is more specific.